### ISSUE TITLE
xarm6 Hand-eye Calibration Demo Result

### ISSUE THREAD

**Issue by KylecheungKml** [xArm-Developer/xarm_ros#183]
We are attempting to do the "7.2 Hand-eye Calibration Demo" using a 3D-print UFACTORY camera stand and compare our output with the sample calibration results provided by you. However, we have observed a difference between our output and the sample calibration results. The hand-eye calibration results seem to translate the camera to the centre location of the end effector at different z positions but not the camera position in the real world. The z-axis distance offset seems correct if 0.15 means 1.5cm.  Can you please provide assistance in resolving this issue? Thanks.

![1](https://user-images.githubusercontent.com/130741691/231986802-9a1b6af5-6c0f-480f-9754-6000e00db3ca.png)
![2](https://user-images.githubusercontent.com/130741691/231986822-8b45ff18-64a7-4aa7-998c-b55365b33b4e.png)
![3](https://user-images.githubusercontent.com/130741691/231986846-742525ed-f07f-44ee-a8f6-01cb8a49a6df.jpg)

**Comment by penglongxiang** [comment]
Hi, it seems your calibration result is off especially in the Z direction. Even our sample result is not 100% accurate, it still serves as a reference and the correct/usable result should not deviate too much from it. Please re-check your whole calibration process and consult "easy_handeye" developer team if necessary.

The camera_color_optical_frame shown in your second picture during calibration process is not from the final result, but a dummy fixed position (only a non-zero z offset) since it is not known before calibration is performed.  The visualization would not be correct unless you publish the calibrated TF relationship (from end_effector to camera) afterwards in your vision application.

**Comment by penglongxiang** [comment]
For the real world position of “camera_color_optical_frame”, please refer to this reply:  https://github.com/xArm-Developer/xarm_ros/issues/132#issuecomment-1272433065

**Comment by KylecheungKml** [comment]
Thank you for your help. After modifying the static_transform_publisher of dummy_handeye in the calibrate.launch file to 0 0 0.05 1.5708 0 0, I was able to achieve a result similar to the sample result shown in RVIZ.
![1](https://user-images.githubusercontent.com/130741691/232703728-0f19a799-ae82-4c47-ad47-944f394e2615.png)

However, when I ran the command "ros2 launch xarm_planner xarm6_planner_realmove.launch.py robot_ip:=192.168.1.237 add_realsense_d435i:=true" in the terminal, I noticed that the camera_color_optical_frame was different from the result obtained in ROS1 and some camera frame is out of the camera model. I am unsure if this is an issue.
![2](https://user-images.githubusercontent.com/130741691/232704576-10577760-b7c2-49a8-9300-04f5e00e5b61.png)

**Comment by KylecheungKml** [comment]
Thank you for your help. After modifying the static_transform_publisher of dummy_handeye in the calibrate.launch file to 0 0 0.05 1.5708 0 0, I was able to achieve a result similar to the sample result shown in RVIZ.
![1](https://user-images.githubusercontent.com/130741691/232705322-e6419307-9135-4e81-90c8-4fcefb2879da.png)


However, when I ran the command "ros2 launch xarm_planner xarm6_planner_realmove.launch.py add_realsense_d435i:=true" in the terminal, I noticed that the camera_color_optical_frame was different from the result obtained in ROS1 and some camera frames are out of the camera model. I am unsure if this is an issue.
![2](https://user-images.githubusercontent.com/130741691/232705449-91fb381d-3e14-4801-a77c-e3e6ca18cd4c.png)
